lib/Sema/DerivedConformanceCodable.cpp

```diff
@@ -89,6 +89,12 @@ static CodableConformanceType typeConformsToCodable(TypeChecker &tc,
   // Some generic types need to be introspected to get at their "true" Codable
   // conformance.
   auto canType = target->getCanonicalType();
+  if (auto referenceType = dyn_cast<ReferenceStorageType>(canType)) {
```

Comment:
@jrose-apple Or are you saying that I never have to get the canonical type to begin with and just use `getAs` on the original target type?